xds/internal/client/attributes.go

```diff
@@ -17,20 +17,34 @@
 
 package client
 
-import "google.golang.org/grpc/resolver"
+import (
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/xds/internal/client/bootstrap"
+	"google.golang.org/grpc/xds/internal/client/load"
+)
 
 type clientKeyType string
 
 const clientKey = clientKeyType("grpc.xds.internal.client.Client")
 
+// Interface contains a subset of the *Client methods that are needed by the
+// balancers.
+type Interface interface {
```

Comment:
I don't know how to name this...

Reply:
Does this contain all of `*xdsclient.Client`'s functions?

If so, unexport `xdsclient.Client`?  If not, maybe add the missing methods & unexport?

If you do that, name this `XDSClient`.  Otherwise, `Interface` is okay.

Reply:
Done

Reply:
Replaced all the testing only interfaces with the real XDSClient interface.